### PR TITLE
[EMCAL-830] Add EMCAL recalibration workflow to async reconstruction

### PR DESCRIPTION
**Pull request by mfasDa**
- CTF decoder publishing publish EMCAL objects on
  subspec 1 in case of AOD production
- Add o2-emcal-cell-recalibrator-workflow in workflows
  producing AODs

### CODE REVIEW

File: prodtests/full-system-test/dpl-workflow.sh [hunk 1]
```diff
@@ -309,7 +309,9 @@ if [[ $CTFINPUT == 1 ]]; then
   [[ ! -z $INPUT_FILE_LIST ]] && CTFName=$INPUT_FILE_LIST
   if [[ -z $CTFName && $WORKFLOWMODE != "print" ]]; then echo "No CTF file given!"; exit 1; fi
   if [[ $NTIMEFRAMES == -1 ]]; then NTIMEFRAMES_CMD= ; else NTIMEFRAMES_CMD="--max-tf $NTIMEFRAMES"; fi
-  add_W o2-ctf-reader-workflow "--delay $TFDELAY --loop $TFLOOP $NTIMEFRAMES_CMD --ctf-input ${CTFName} ${INPUT_FILE_COPY_CMD+--copy-cmd} ${INPUT_FILE_COPY_CMD} --onlyDet $WORKFLOW_DETECTORS --pipeline $(get_N tpc-entropy-decoder TPC REST 1 TPCENTDEC)"
+  CTF_EMC_SUBSPEC=
+  workflow_has_parameter AOD && has_detector && [[ ! -z "$AOD_INPUT" ]] && CTF_EMC_SUBSPEC="--emcal-decoded-subspec 1"
```
Comment (shahor02) [inline prodtests/full-system-test/dpl-workflow.sh]: should be `has_detector EMC`

File: prodtests/full-system-test/dpl-workflow.sh [hunk 2]
```diff
@@ -309,7 +309,9 @@ if [[ $CTFINPUT == 1 ]]; then
...
+  add_W o2-ctf-reader-workflow "--delay $TFDELAY --loop $TFLOOP $NTIMEFRAMES_CMD --ctf-input ${CTFName} ${INPUT_FILE_COPY_CMD+--copy-cmd} ${INPUT_FILE_COPY_CMD} --onlyDet $WORKFLOW_DETECTORS --pipeline $(get_N tpc-entropy-decoder TPC REST 1 TPCENTDEC) $CTF_EMC_SUBSPEC"
```
Comment (shahor02) [inline prodtests/full-system-test/dpl-workflow.sh]: yes, but please move `$CTF_EMC_SUBSPEC` before the `--pipeline...` to have uniform patter for all workflows.

File: prodtests/full-system-test/dpl-workflow.sh [hunk 3]
```diff
@@ -485,6 +487,7 @@ workflow_has_parameter GPU_DISPLAY && [[ $NUMAID == 0 ]] && add_W o2-gpu-display
 # AOD
 [[ -z "$AOD_INPUT" ]] && AOD_INPUT=$TRACK_SOURCES
 ( ! has_detector_matching SECVTX || ! has_detectors_reco ITS || [[ $BEAMTYPE == "cosmic" ]]) && AODPROD_OPT+="--disable-secondary-vertices"
+workflow_has_parameter AOD && has_detector && [[ ! -z "$AOD_INPUT" ]] && add_W o2-emcal-cell-recalibrator-workflow "--input-subspec 1 --output-subspec 0"
```
Comment (shahor02) [inline prodtests/full-system-test/dpl-workflow.sh]: here too: `has_detector EMC`

**Review by shahor02 — CHANGES_REQUESTED** [review]

**Comment by mfasDa** [comment]
I had to 2 two changes in order to be able to start the workflow:
- Condition on $AOD_INPUT had to be remove, otherwise the selections were false, obviously $AOD_INPUT was empty
- The recalibration workflow must be defined after before the primary vertex workflow, otherwise the primary vertex workflow does not find the outputs produced by the EMCAL cell recalibration workflow

**Comment by mfasDa** [comment]
I tested the workflow offline using the DPG settings for LHC22f and an example run from LHC22m (521348) for which the calibration was uploaded to the CCDB and the cells from the EMCAL side looks good, data was properly calibrated.

Before merging we need the default gain calibration params to be uploaded to the CCDB (https://alice.its.cern.ch/jira/browse/O2-3130), otherwise the workflow will crash due to missing calibration object.

**Comment by mfasDa** [comment]
Full CI is crashing in Fatal because it doesn't find calibration objects for unanchored MC. The porting request ics created here: https://alice.its.cern.ch/jira/browse/O2-3209. Once the CCDB objects are available merging can go ahead.

**Comment by davidrohr** [comment]
looks good to me now

**Comment by mfasDa** [comment]
Linking ticket https://alice.its.cern.ch/jira/browse/EMCAL-837 of the change based on an offline reconstruction of run 521348 (LHC22m) where one can see that the signal is nicely calibrated now and the EMCAL cells in the AOD are consistent with the output of the recalibration workflow.

**Comment by mfasDa** [comment]
Last crash in the full CI was due to missing gain calibration object for the unanchored MC range. This is now uploaded as well (https://alice.its.cern.ch/jira/browse/O2-3130). Once the CI finishes it should be ready for merge.

**Review by shahor02 — APPROVED** [review]

**Pull request merged**